### PR TITLE
[BugFix]modify first_value for unbounded preceding and current row

### PR DESCRIPTION
**Pull request by before-Sunrise**
Signed-off-by: before-Sunrise <unclejyj@gmail.com>

## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
- [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [x] 2.4
  - [x] 2.3
  - [x] 2.2

### CODE REVIEW

**Comment by before-Sunrise** [comment]
@Mergify backport  branch-2.2 branch-2.3 branch-2.4 branch-2.5

File: be/src/exprs/agg/window.h [hunk 1]
```diff
@@ -324,6 +326,13 @@ class FirstValueWindowFunction final : public ValueWindowFunction<PT, FirstValue
             return;
         }
 
+        // only calculate once
+        if (this->data(state).count != 0 && (!this->data(state).is_null || !ignoreNulls)) {
```
Comment (liuyehcf) [inline be/src/exprs/agg/window.h:330]: what if ignoreNulls=true, and this->data(state).is_null = true

**Review by liuyehcf — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.3

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.2

**Pull request merged**